### PR TITLE
feat(modals, notifications): Add animation support

### PR DESCRIPTION
**Pull request by kgalanop**
This is an initial support for animations after updating to Angular 4. Currently modals and notifications are supported.

### CODE REVIEW

**Comment by kgalanop** [comment]
Thank you @esarbanis for your comments. I will check if I can create an extra input for each component, to enable/disable animations, integrating the newly created attribute @.disabled (https://github.com/angular/angular/issues/16483)

**Comment by bekos** [comment]
I think the correct way to handle it is like this: https://github.com/angular/material2/blob/1af2ce6e1797a05934f1d30b9c1fc1b625d68bcf/guides/getting-started.md#step-2-animations

**Comment by esarbanis** [comment]
@bekos I would argue that it would be extra useful to have our configuration setup animations or not.

Use cases:
- The user imports `BrowserAnimationsModule` in her own custom components and doesn't want to use animations in ngl. In that case she should be able to completely disable animations in `NglConfig`.
- The user imports `BrowserAnimationsModule` and wan't to use animations in ngl except let's say in the modals or in the notifications. In that case she should be able to disable animations in the component context (`[animation]="false"`?).
- The user does not want animations at all so she imports `NoopAnimationsModule`. We got that covered by the current implementation :wink:

File: src/modals/modal.ts [hunk 1]
```diff
@@ -45,14 +60,30 @@ export class NglModal {
...
   @HostListener('keydown.esc', ['$event'])
   close(evt?: Event) {
+    this._animationOpen = false;
     if (evt) {
       evt.stopPropagation();
     }
-    this.openChange.emit(false);
+  }
+
+  @HostListener('click', ['$event'])
```
Comment (ihym) [inline src/modals/modal.ts]: Is this necessary? #341

**Comment by kgalanop** [comment]
@ihym , you are right, there is no use for this line and it has been fixed.
Regarding the animations functionality, I believe that we could keep this in the same way as described in https://github.com/angular/material2/blob/1af2ce6e1797a05934f1d30b9c1fc1b625d68bcf/guides/getting-started.md#step-2-animations too and used in other implementations as well. As this is a new feature, as long as there are no other issues, I wouldn't regard this as blocking for the merge. User can enable/disable animations in that way. 

Nevertheless, I will check how we could enable animations as an input property for each component as an improvement of the implemented feature.

**Comment by bekos** [comment]
Unfortunately, I am not very familiar with the animations in Angular and not very confident that I will be able to maintain correctly such functionality, neither that I will be able to provide the correct guidelines. So, as long as we don't have someone really dedicated to this project and with expertise into Angular animations, I prefer to close this PR.

@kgalanop I am really sorry for this and really appreciate your effort and your contributions until know, and I hope that you can understand that there are a lot of issues to handle already, so adding animations at this moment would be overkill. Although we weren't very consistent, we think this has changed and hope you will be able to contribute in the future ;)

**Pull request closed without merging**